lib/hal_client.rb

```diff
@@ -119,7 +124,14 @@ def def_unsafe_request(method)
 
         begin
           client = client_for_post(override_headers: headers)
-          resp = bmtb("#{verb} <#{url}>") { client.request(method, url, body: req_body) }
+          request_lambda = -> { client.request(method, url, body: req_body) }
+          resp = bmtb("#{verb} <#{url}>") {
+            if retryable
+              retryinator.call { request_lambda.call }
+            else
+              request_lambda.call
+            end
+          }
```

Comment:
perhaps it would be better to have separate `#def_idempotent_unsafe_request` and `#def_unsafe_request` methods? doing so would allow this mess to be separated out into the logically distinct parts that it actually is.

Reply:
agreed